services/frontend-service/src/ui/ActionDetails.tsx

```diff
@@ -82,104 +118,181 @@ type ActionDetails = {
-                type: ActionTypes.DeleteApplicationLock,
-                name: 'Delete App Lock',
-                dialogTitle: 'Are you sure you want to delete this application lock?',
-                summary:
-                    'Unlock "' +
-                    action.action?.deleteEnvironmentApplicationLock.application +
-                    '" on ' +
-                    action.action?.deleteEnvironmentApplicationLock.environment,
-                icon: <LockOpenRounded />,
-                environment: action.action?.deleteEnvironmentApplicationLock.environment,
-                application: action.action?.deleteEnvironmentApplicationLock.application,
-                lockId: action.action?.deleteEnvironmentApplicationLock.lockId,
-            };
-        case 'prepareUndeploy':
-            return {
-                type: ActionTypes.PrepareUndeploy,
-                name: 'Prepare Undeploy',
-                dialogTitle: 'Are you sure you want to start undeploy?',
-                description:
-                    'The new version will go through the same cycle as any other versions' +
-                    ' (e.g. development->staging->production). ' +
-                    'The behavior is similar to any other version that is created normally.',
-                summary: 'Prepare undeploy version for Application ' + action.action?.prepareUndeploy.application,
-                icon: <DeleteOutlineRounded />,
-                application: action.action?.prepareUndeploy.application,
-            };
-        case 'undeploy':
-            return {
-                type: ActionTypes.Undeploy,
-                name: 'Undeploy',
-                dialogTitle: 'Are you sure you want to undeploy this application?',
-                description: 'This application will be deleted permanently',
-                summary: 'Undeploy and delete Application ' + action.action?.undeploy.application,
-                icon: <DeleteForeverRounded />,
-                application: action.action?.undeploy.application,
-            };
-        default:
-            return {
-                type: ActionTypes.UNKNOWN,
-                name: 'invalid',
-                dialogTitle: 'invalid',
-                summary: 'invalid',
-                icon: <Error />,
-            };
-    }
+    if ('deploy' in action)
+        return {
+            type: ActionTypes.Deploy,
+            name: 'Deploy',
+            dialogTitle: 'Are you sure you want to deploy this version?',
+            summary:
+                'Deploy version ' +
+                action.deploy.version +
+                ' of "' +
+                action.deploy.application +
+                '" to ' +
+                action.deploy.environment,
+            icon: <MoveToInboxRounded />,
+            environment: action.deploy.environment,
+            application: action.deploy.application,
+            version: action.deploy.version,
+        };
+    if ('createEnvironmentLock' in action)
```

Comment:
Since `switch` in JavaScript cannot have conditions in `case` (which is what I would have used in Golang in this case), I would use `else if` here to signal that this is one large block that belongs together and examines `action`:
```suggestion
    else if ('createEnvironmentLock' in action)
```